pkg/statistics/cmsketch.go

```diff
@@ -259,10 +260,15 @@ func (c *CMSketch) SubValue(h1, h2 uint64, count uint64) {
 // QueryValue is used to query the count of specified value.
 func QueryValue(sctx sessionctx.Context, c *CMSketch, t *TopN, val types.Datum) (uint64, error) {
 	var sc *stmtctx.StatementContext
+	tz := time.UTC
 	if sctx != nil {
 		sc = sctx.GetSessionVars().StmtCtx
+		tz = sc.TimeZone()
+	}
+	rawData, err := tablecodec.EncodeValue(tz, nil, val)
+	if sc != nil {
+		err = sc.HandleError(err)
```

Comment:
I don't think we need `if sc != nil` because it is checked in `sc.HandleError`.

Reply:
I personally hope every call to the `sc.HandleError` can ensure that the `sc != nil`. I think as we continue on refractoring the stmt context, we can finally reach the goal, but not now.

The `if sc != nil` in `sc.HandleError` is there only to avoid panic in production. We also have a `intest.AssertNotNil(sc)` to assert it's not nil in tests. We have done similar things in `sc.TimeZone()`.

Reply:
OK.